parsing/parser.mly

```diff
@@ -2483,22 +2483,22 @@ label_declaration_semi:
 
 str_type_extension:
   TYPE ext_attributes
-  nonrec_flag optional_type_parameters mkrhs(type_longident)
+  /* NONREC not allowed */
+  optional_type_parameters mkrhs(type_longident)
   PLUSEQ private_flag str_extension_constructors post_item_attributes
       { let (ext, attrs) = $2 in
-        if $3 <> Recursive then not_expecting $loc($3) "nonrec flag";
```

Comment:
Before this change, the error message was as follows:

```
# module type M = sig type nonrec t += A end;;
                           ^^^^^^
Error: Syntax error: nonrec flag not expected.
```


After this change, the error message is as follows:

```
# module type M = sig type nonrec t += A end;;
                                    ^^
Error: Syntax error: 'end' expected
Line 1, characters 16-19: This 'sig' might be unmatched
```